server/auth/auth.go

```diff
@@ -790,14 +790,50 @@ const (
 
 // UserSettlingLimit returns a user's settling amount limit for the given market
 // in units of the base asset. The limit may be negative for accounts with poor
-// swap history.
+// swap history. The user must be connected.
 func (auth *AuthManager) UserSettlingLimit(user account.AccountID, mkt *dex.MarketInfo) int64 {
+	// WIP: https://github.com/orgs/decred/projects/2?pane=issue&itemId=2420755
+	// tier is a balance of bonding strength (bondTier) and user score, where
+	// the latter is computed from all violations and match outcomes. See the
+	// integrateOutcomes and tier methods. However, for order size limits, we
+	// consider swap outcome amounts, and consider only swap outcomes (ignore
+	// preimage and cancel rate violations). Thus, we scale the base limit by
+	// bond strength, and balance that with weighted swap history.
+
 	currentLotSize := int64(mkt.LotSize)
-	base := currentLotSize * auth.initTakerLotLimit
+	baseLimit := currentLotSize * auth.initTakerLotLimit
+
+	// The base limit scales with bond strength (plus one if legacy fee paid).
+	// This is particularly important for an account that has offset their
+	// violations with additional bond to gain a positive tier.
+	client := auth.user(user)
+	if client == nil {
+		return 0 // offline
+	}
+	client.mtx.Lock()
+	tier := client.tier
+	baseScale := client.bondTier()
+	if client.legacyFeePaid {
+		baseScale++
+	}
+	client.mtx.Unlock()
+
+	if tier < 1 { // but order router should have checked this
+		return 0
+	}
```

Comment:
Can we not return a limit for an online user that has paid a legacy fee but currently have zero tiers?

Reply:
Zero tier means not allowed to trade, by definition.  The incoming order processing pipeline should already have prevented this, but it's here for good measure.

Legacy fee users get a tier increment already, so if such a user is not positive tier, that means they have to many violations.

Reply:
> Legacy fee users get a tier increment already, so if such a user is not positive tier, that means they have too many violations.

Okay, thanks for the clarification.